derive-macros/src/lib.rs

```diff
@@ -84,23 +92,21 @@ fn gen_schema_fields(data: &Data) -> TokenStream {
                     match &segment.arguments {
                         PathArguments::None => quote! { #segment_ident :: },
                         PathArguments::AngleBracketed(angle_args) => quote! { #segment_ident::#angle_args :: },
-                        _ => panic!("Can only handle <> type path args"),
+                        _ => Error::new(segment.arguments.span(), "Can only handle <> type path args").to_compile_error()
                     }
                 });
                 if have_schema_null {
                     if let Some(first_ident) = type_path.path.segments.first().map(|seg| &seg.ident) {
                         if first_ident != "HashMap" {
-                            panic!("Can only use drop_null_container_values on HashMap fields, not {first_ident:?}");
+                           return Error::new(first_ident.span(), "Can only use drop_null_container_values on HashMap fields, not {first_ident:?}").to_compile_error()
```

Comment:
`panic!` formats the args, but the error class doesn't.

```suggestion
                           return Error::new(
                                first_ident.span(),
                                format!("Can only use drop_null_container_values on HashMap fields, not {first_ident}")
                            ).to_compile_error()
```

Reply:
ahh, yes, old habits. I'm working on tests for these, should be easier to spot